<launch>
  <arg name="xtion" default= "false"/>
  <arg name="usb_cam" default= "false"/>
  <arg name="frame_id" default="/kinect_optical_frame"/>

  <include file="$(find state_manager)/launch/stateManager.launch" />
  <include file="$(find pandora_vision_qrcode)/launch/pandora_vision_qrcode.launch"
    ns="/pandora_vision/pandora_vision_qrcode" >
    <arg name="standalone_nodelet" value="true" />
  </include>

  <include file="$(find pandora_robots_description)/gio/launch/gio_description.launch" />

  <group if="$(arg xtion)">
    <param name="kinect/topic_name" value="/kinect/rgb/image_raw" />
    <param name="dataset_path" value="/test/benchmark/images/images_480p"/>
    <param name="dataset_camera" value="kinect"/>
    <param name="/kinect_optical_frame/hfov" value="58.0" />
    <param name="/kinect_optical_frame/vfov" value="45.0" />
  </group>

 <group if="$(arg usb_cam)">
    <param name="kinect/topic_name" value="/left_camera/image" />
    <param name="dataset_path" value="/test/benchmark/images/images_720p"/>
    <param name="dataset_camera" value="web_camera"/>
    <param name="frame_id" value="left_camera_optical_frame" />
    <param name="$(arg frame_id)/hfov" value="60.00" />
    <param name="$(arg frame_id)/vfov" value="33.75" />
  </group>

  <param name="benchmarkFlag" value="False" />
  <param name="publisherMessagePackage" value="sensor_msgs" />
  <param name="publisherMessageType" value="Image"/>

  <param name="subscriberTopic" value="/pandora_vision/pandora_vision_qrcode/pandora_qrcode/processor_log" />
  <param name="subscriberMessagePackage" value="sensor_processor" />
  <param name="subscriberMessageType" value="ProcessorLogInfo"/>

  <param name="subscriberAlertTopic" value="/vision/qr_alert" />
  <param name="subscriberAlertMessagePackage" value="pandora_vision_msgs" />
  <param name="subscriberAlertMessageType" value="QRAlertVector"/>

  <param name="algorithm" value="QR"/>

  <test test-name="QRBenchmarkTest" pkg="pandora_vision_qrcode"
    type="qrcode_benchmark_test.py" time-limit="2000.0"/>

</launch>
